runtime/sema/check_composite_declaration.go

```diff
@@ -1030,18 +1030,29 @@ func (checker *Checker) initializerParameters(initializers []*ast.SpecialFunctio
 
 	initializerCount := len(initializers)
 	if initializerCount > 0 {
+
 		firstInitializer := initializers[0]
+
 		parameters = checker.parameters(firstInitializer.FunctionDeclaration.ParameterList)
 
 		if initializerCount > 1 {
+
+			firstInitializer := initializers[0]
 			secondInitializer := initializers[1]
 
+			previousPos := firstInitializer.StartPosition()
+
+			pos := secondInitializer.StartPosition()
+
 			checker.report(
-				&UnsupportedOverloadingError{
-					DeclarationKind: common.DeclarationKindInitializer,
-					Range:           ast.NewRangeFromPositioned(checker.memoryGauge, secondInitializer),
+				&RedeclarationError{
+					Kind:        common.DeclarationKindInitializer,
+					Name:        "",
```

Comment:
Fill out the name here

```suggestion
					Name:        common.DeclarationKindInitializer.Keywords(),
```